include/nuttx/tee.h

```diff
@@ -0,0 +1,479 @@
+/****************************************************************************
+ * include/nuttx/tee.h
+ *
+ * Copyright (c) 2015-2016, Linaro Limited
```

Comment:
This is required to be documented in [LICENSE](https://github.com/apache/nuttx/blob/master/LICENSE).

Also, BSD-licensed code should be gated behind `ALLOW_BSD_COMPONENTS`